comp/forwarder/defaultforwarder/resolver/domain_resolver.go

```diff
@@ -33,38 +38,111 @@ type DomainResolver interface {
 	// Resolve returns the domain to be used to send data for a given `transaction.Endpoint` along with a
 	// destination type
 	Resolve(endpoint transaction.Endpoint) (string, DestinationType)
+	// GetAPIKeysInfo returns the list of API Keys and config paths associated with this `DomainResolver`
+	GetAPIKeysInfo() []utils.APIKeys
 	// GetAPIKeys returns the list of API Keys associated with this `DomainResolver`
 	GetAPIKeys() []string
+	// SetAPIKeys updates the deduped list of API keys associated with this `DomainResolver`
+	// Returns the list of keys that have changed - the returned keys are scrubbed as these will be used to log
+	// which keys have been changed.
```

Comment:
Not a huge deal, but I think this is suboptimal abstraction: the resolver shouldn't be tasked with preparing api keys for logging if it isn't doing the logging itself, and it causes code the same bit of code to be duplicated that could be just written once inline with the logging call once.